pennylane/pauli/grouping/group_observables.py

```diff
@@ -112,68 +145,260 @@ def binary_repr(self, n_qubits=None, wire_map=None):
         else:
             self._wire_map = wire_map
 
-        self._n_qubits = n_qubits
-
         return observables_to_binary_matrix(self.observables, n_qubits, self._wire_map)
 
-    def complement_adj_matrix_for_operator(self):
-        """Constructs the adjacency matrix for the complement of the Pauli graph.
+    @cached_property
+    def adj_matrix(self) -> np.ndarray:
+        """Adjacency matrix for the complement of the Pauli graph determined by the ``grouping_type``.
 
-        The adjacency matrix for an undirected graph of N vertices is an N by N symmetric binary
+        The adjacency matrix for an undirected graph of N nodes is an N by N symmetric binary
         matrix, where matrix elements of 1 denote an edge, and matrix elements of 0 denote no edge.
+        """
+        return adj_matrix_from_symplectic(self.binary_observables, grouping_type=self.grouping_type)
+
+    def colour_pauli_graph(self):
+        """
+        Runs the graph colouring heuristic algorithm to obtain the partitioned Pauli observables.
 
         Returns:
-            array[int]: the square and symmetric adjacency matrix
+            list[list[Observable]]: List of partitions of the Pauli observables made up of mutually (anti-)commuting observables.
         """
+        if self.graph_colourer == "rlf":
+            coloured_binary_paulis = recursive_largest_first(
+                self.binary_observables, self.adj_matrix
+            )
 
-        if self.binary_observables is None:
-            self.binary_observables = self.binary_repr()
+            # Need to convert back from the symplectic representation
+            pauli_groups = [
+                [binary_to_pauli(pauli_word, wire_map=self._wire_map) for pauli_word in grouping]
+                for grouping in coloured_binary_paulis.values()
+            ]
 
-        n_qubits = int(np.shape(self.binary_observables)[1] / 2)
+        else:
+            pauli_groups = self.pauli_partitions_from_graph()
 
-        if self.grouping_type == "qwc":
-            adj = qwc_complement_adj_matrix(self.binary_observables)
+        return pauli_groups
 
-        elif self.grouping_type in frozenset(["commuting", "anticommuting"]):
-            symplectic_form = np.block(
-                [
-                    [np.zeros((n_qubits, n_qubits)), np.eye(n_qubits)],
-                    [np.eye(n_qubits), np.zeros((n_qubits, n_qubits))],
-                ]
-            )
-            mat_prod = (
-                self.binary_observables @ symplectic_form @ np.transpose(self.binary_observables)
+    def idx_partitions_from_graph(self) -> list[list]:
```

Comment:
Since this is being used outside of the class in: 

https://github.com/PennyLaneAI/pennylane/blob/b28c41a017bc85d816b89634d5dadd5731337824/pennylane/pauli/grouping/group_observables.py

I guess there is no need to make it private, right?